environment.yml

```diff
@@ -70,9 +70,9 @@ dependencies:
   # benchmarks
   - asv>=0.5.1
 
-  # The compiler packages are meta-packages and install the correct compiler (activation) packages on the respective platforms.
-  - c-compiler
-  - cxx-compiler
+  ## The compiler packages are meta-packages and install the correct compiler (activation) packages on the respective platforms.
+  #- c-compiler
```

Comment:
Is this linked to a particular bug in meson? Worried about taking this out of environment.yml for new contributors

Reply:
On macOS, I am segfaulting on import with the c/cxx-compiler packages. 

(I'm getting a warning about ld not being able to respect the -pie flag for a shared library, so I guess something about the way meson is handling position independent code?)

Reply:
I think pie is just for executables. So not sure why that would make its way into the library linking commands. Think that is a bug with our config or within meson?

Reply:
`-pie` warnings come from the conda-forge compilers, because they are added to CFLAGS/LDFLAGS by the compiler activation scripts. It's harmless AFAIK, but quite verbose and annoying.

Reply:
Ah interesting. Maybe it's masking the real issue then. If you attach Python to gdb it should open up the debugger at the moment the segfault occurs

Reply:
Super strange. Can you share the backtrace? Frames are likely going to have limited information within them from it being a non-debug build but the backtrace might clue us in. 

Ping @jorisvandenbossche for any thoughts on the conda compilers

Reply:
Also if you can share the gcc commands used to build the extensions would be helpful

Reply:
Last thing I can think to try is changing the LDFLAGS to not add the -fpie as @rgommers pointed out and see if that helps

Reply:
> `-pie` warnings come from the conda-forge compilers, because they are added to CFLAGS/LDFLAGS by the compiler activation scripts. It's harmless AFAIK, but quite verbose and annoying.

Thanks for the tip here. I went and echo'ed LDFLAGS, which resulted in
```-Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/thomasli/opt/miniconda3/envs/pandas-dev2/lib -L/Users/thomasli/opt/miniconda3/envs/pandas-dev2/lib```.

I think the last part is what's causing the issue.

Reply:
Does doing `LDFLAGS=${LDFLAGS}, -Wl,-no-pie pip install ...` help at all?

Reply:
I only use the conda compilers, because for SciPy we need `gfortran` anyway and there's no such thing from Apple (and just `conda install compilers` is nicer than pulling in Homebrew). Apple compilers should work too though, I know that there are folks using those with SciPy, so the problem is likely Pandas-specific (or macOS-version specific, or ...).

I have an M1 Macbook, curious if I can reproduce it. Is it a matter of removing `c-compiler` and `cxx-compiler` from `environment.yml`, creating a new env with that, and building from this branch?

Reply:
> I only use the conda compilers, because for SciPy we need `gfortran` anyway and there's no such thing from Apple (and just `conda install compilers` is nicer than pulling in Homebrew). Apple compilers should work too though, I know that there are folks using those with SciPy, so the problem is likely Pandas-specific (or macOS-version specific, or ...).
> 
> I have an M1 Macbook, curious if I can reproduce it. Is it a matter of removing `c-compiler` and `cxx-compiler` from `environment.yml`, creating a new env with that, and building from this branch?

For me, removing c-compiler and cxx-compiler make it work. I've verified that the issue is not with pie (I think it worked when I set LDFLAGS to just ``-Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl``). 

It might be an issue of linking incorrectly against Python (I suspect the last ``-L <conda env lib dir>``).